lib/ansible/plugins/lookup/varnames.py

```diff
@@ -0,0 +1,79 @@
+# (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+    lookup: varnames
+    author: Ansible Core
+    version_added: "2.8"
+    short_description: Lookup matching variable names
+    description:
+      - Retrieves a list of matching Ansible variable names.
+    options:
+      _terms:
+        description: List of patterns to match to variable names.
+        required: True
+"""
+
+EXAMPLES = """
+- name: List variables that start with qz_
+  debug: msg="{{ lookup('varnames', 'qz_.+')}}"
+  vars:
+    qz_1: hello
+    qz_2: world
+    qa_1: i wont show
+    qz_: i wont show either
+
+- name: Show all variables
+  debug: msg="{{ lookup('varnames', '.+')}}"
+
+- name:  show variables with 'hosts' in their names
+  debug: msg="{{ lookup('varnames', '.*hosts.*')}}"
```

Comment:
```suggestion
  debug: msg="{{ lookup('varnames', 'hosts')}}"
```
should also do the same thing.

Or do you want to use `name.match()` instead of `name.search()` below?